src/AutoRest.CSharp/LowLevel/Output/LowLevelOutputLibraryFactory.cs

```diff
@@ -229,7 +238,7 @@ private static IEnumerable<LowLevelClient> CreateClients(IEnumerable<ClientInfo>
         public class ClientInfo
         {
             public string OperationGroupKey { get; }
-            public string Name { get; }
+            public string Name { get; set; }
```

Comment:
This property should always remain read-only because it is used as a key in a dictionary:
```cs
var clientInfosByName = codeModel.OperationGroups
    .Select(og => CreateClientInfo(og, context))
    .ToDictionary(ci => ci.Name);
```

In general, it is a good idea to check how read-only property is used before making it writable.

Reply:
Thank @AlexanderSher But now we have problem, usually we need to modify the clientName if we want to support both 'single-top-level-client' and `client hierarchy`. If Name is readonly, we need to create an new object instance when we want to update. Maybe we need to define other readonly field to use as key of dictionary.